src/controllers/destination.controller.ts

```diff
@@ -0,0 +1,84 @@
+import { Context } from "koa";
+import PreTransformationServiceDestination from "../services/destination/preTransformation.destination.service";
+import { MiscService } from "../services/misc.service";
+import { ProcessorRequest, RouterRequestData, RouterRequest } from "../types/index";
+import { ServiceSelector } from "../util/serviceSelector";
+
+export default class DestinationController {
+  public static async destinationTransformAtProcessor(ctx: Context) {
+    let requestMetadata = MiscService.getRequestMetadata(ctx);
+    let events = ctx.request.body as ProcessorRequest[];
+    const {
+      version,
+      destination
+    }: { version: string; destination: string } = ctx.params;
+    const integrationService = ServiceSelector.getDestinationService(events);
+    const destinationHandler = ServiceSelector.getDestHandler(
+      destination,
+      version
+    );
+    events = PreTransformationServiceDestination.preProcess(
+      events,
+      ctx
+    ) as ProcessorRequest[];
+    const resplist = await integrationService.processorRoutine(
+      events,
+      destination,
+      destinationHandler,
+      requestMetadata
+    );
+    ctx.body = resplist;
+    MiscService.transformerPostProcessor(ctx);
```

Comment:
Theoretically, it's not a service's responsibility to update the `ctx` object.

Reply:
Actually this is a post processing layer not sure what to call this

Reply:
This can be a utility method within the controllers.

Reply:
Done